feat: add support for stdin and stdin-filename options

SirSam21 opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests.
-->

<!-- _Please make sure to review and check all of these items:_ -->

Closes #803

#### ✔ Checklist:

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ ] Commit messages have been written in [Conventional Commits](https://www.conventionalcommits.org) format;
- [x] Tests for the changes have been added (for bug fixes / features);
- [ ] Docs have been added / updated on the [dotenv-linter.github.io](https://github.com/dotenv-linter/dotenv-linter.github.io) (for bug fixes / features).

<!-- _NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._ -->

SirSam21 committed:
added stdin arg

SirSam21 committed:
stdin refactor and new optional stdin_filename arg

refactored stdin flag to align more with repo coding style, added stdin_filename flag.

SirSam21 committed:
Added tests for stdin and stdin-filename options, fixed some bugs

SirSam21 commented:
I wanted to add that some of this code may look awkward because I tried to make stdin fit into the project structure that focuses on traversing directories and reading files directly. I'm hoping to get some input from others on how this might be addressed.

DDtKey commented on dotenv-linter/src/cli.rs:
```diff
@@ -139,6 +139,15 @@ fn common_args() -> Vec<Arg> {
+            .long("stdin")
+            .help("Read and lint from stdin instead of files")
+            .action(ArgAction::SetTrue),
+        Arg::new("stdin-filename")
+            .long("stdin-filename")
+            .default_value("")
+            .help("Specify file name to process stdin as")
+            .action(ArgAction::Set),
```
I think we have decided not to support this option. So we can drop it

DDtKey commented on dotenv-linter/src/cli.rs:
```diff
@@ -139,6 +139,15 @@ fn common_args() -> Vec<Arg> {
             .long("schema")
             .help("Use schema file to check .env files")
             .value_parser(value_parser!(PathBuf)),
+        Arg::new("stdin")
+            .long("stdin")
+            .help("Read and lint from stdin instead of files")
+            .action(ArgAction::SetTrue),
```
Just a note: it still makes sense to support special filename (`-`) passed as it was discussed here: https://github.com/dotenv-linter/dotenv-linter/issues/803#issuecomment-2294979944 (and see the next comment)

But not necessary to mix this in single PR for sure. Just something nice to keep in mind

This may help to build something more aligned, e.g unified handling for `-` and `--stdin`

DDtKey reviewed (commented):
Thank you for the contribution! 🚀 

I'll review this in more detail later, but I've left a few preliminary comments to align the expectations for this feature.

SirSam21 committed:
fixed tests, removed --stdin-filename tests

SirSam21 committed:
remove --stdin-filename option

SirSam21 commented:
I removed --stdin-filename and fixed the tests I made. For the tests I pretty much just had to add the with_default_args() to avoid the update check during testing.
